cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1367,6 +1377,7 @@ private void interBrokerMoveReplicas() throws InterruptedException, ExecutionExc
       }
       // At this point it is guaranteed that there are no in execution tasks to wait -- i.e. all tasks are completed or dead.
       if (_stopSignal.get() == NO_STOP_EXECUTION) {
+        updatePartitionMovementMetrics(numTotalPartitionMovements, totalDataToMoveInMB, System.currentTimeMillis() - startTime);
```

Comment:
I think both metrics should be updated in the loop and not just at the end when all is complete. There are already `numFinishedPartitionMovements` and `finishedDataMovementInMB` which can be used to update the gauge. This will give us better insight in the variability of the partition movement speed at a minute granularity. What do you think?

Reply:
I have considered that path. The problem of that is the value of first several loops can be misleading.

E.g. it can take 5 seconds to finish the first 100 partition movement, the the value will be 100/5 = 20. However for the total rebalance process, the average partition movement per sec can be only 0.01.

I believe the average speed of the whole rebalance time is more representative.

Reply:
This chart updated at the end of the rebalance is gonna look like a step function that changes only once a hour or something---when the next rebalance finishes, it will suddenly drop or increase. It's misleading. The correct average rebalance should be some sort of trailing X minutes rebalance speed. will work especially well for the bandwidth chart.   Ingraphs has support for that: https://iwww.corp.linkedin.com/wiki/cf/display/SOP/Ingraphs+Examples#IngraphsExamples-Smoothingbyaveragingthelast30minutesofdata

Reply:
If I understand you right, the value of first several loops is immaterial because the sensor framework will snapshot the value only once a minute. All intermediate values between two snapshots are overwritten. In that sense, the sensor should record only the cumulative time of the finished movements count and data in MB. It will smooth out towards the end of the rebalance towards the average value automatically. May be we don't need trailing x minutes then.

Reply:
Ok I can see either way has some misleading points, but I still think we should only use the overall rebalance speed.

What we need to is a reflection of overall rebalance performance, rather than the rebalance throughput at a certain time. Think about this case: at some cycles, the partition movement can be much faster/slower than other cycles, but we don't care about them. 

In general, I think we should rule out the throughput change during the rebalance, and only emit the value at the end of the rebalance.

Reply:
Consider using 1-, 5-, and 15- minute average when reporting. It will smooth out the bandwidth signal over the last x minutes. For long running rebalances it will be a meaningful signal. For very short running rebalances not so much. but then short running rebalances are not a problem. 

See dropwizard: https://metrics.dropwizard.io/4.2.0/getting-started.html#gauges